app/Filters/Throttle.php

```
@@ -0,0 +1,54 @@
+/**
+ * Rate limits login/migrate POST attempts, keyed by IP and by submitted
+ * username, to mitigate brute-force and credential-stuffing attacks
+ * (GHSA-hm9c-xchj-xgcp). Backed by CodeIgniter's cache-based Throttler,
+ * so limits are per-server (not shared across nodes on file cache).
+ */
+class Throttle implements FilterInterface
+{
+    private const CAPACITY = 5;
+    private const SECONDS  = 60;
+
+    public function before(RequestInterface $request, $arguments = null)
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$arguments'. (undefined)

(UnusedFormalParameter)